cmd/dlv/cmds/commands.go

```diff
@@ -477,20 +481,20 @@ func traceCmd(cmd *cobra.Command, args []string) {
 					return 1
 				}
 				if traceTestBinary {
-					if err := gobuild.GoTestBuild(debugname, dlvArgs, BuildFlags); err != nil {
+					if err := gobuild.GoTestBuild(debugname, DlvArgs, BuildFlags); err != nil {
 						fmt.Fprintf(os.Stderr, "%v\n", err)
 						return 1
 					}
 				} else {
-					if err := gobuild.GoBuild(debugname, dlvArgs, BuildFlags); err != nil {
+					if err := gobuild.GoBuild(debugname, DlvArgs, BuildFlags); err != nil {
 						fmt.Fprintf(os.Stderr, "%v\n", err)
 						return 1
 					}
 				}
 				defer gobuild.Remove(debugname)
 			}
 
-			processArgs = append([]string{debugname}, targetArgs...)
+			processArgs = append([]string{debugname}, TargetArgs...)
```

Comment:
(I can't add a review comment on the appropriate line cause it wasn't changed so I'll add it here)

You must fill in the new struct members you added to `service.Config` below (`Packages` and `BuildFlags`). Currently they are not being passed through and are always empty, it appears.